### PR TITLE
Added context menu hooks to deploy Sam application (actual deploy not implemented)

### PR DESCRIPTION
**Pull request by awschristou**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Description

Uses context menus as the UI entry point for users to publish their Sam Applications to AWS.
Context menu items added to template.yaml/template.yml files in the following locations:
* right clicking on the file in project explorer
* right clicking on the file's editor tab
* right clicking within the open file's editor

The actual publish implementation is covered in #395 

image
image
image

## Related Issue(s)

#390 & #392 

## Testing

* Right click on template.yaml file in Project Explorer, expect to see menu item
* Right click on template.yml file in Project Explorer, expect to see menu item
* Right click on non template.yaml file in Project Explorer, expect to not see menu item
* Open template.yaml file, right click on editor tab, expect to see menu item
* Open template.yaml file, right click within file, expect to see menu item
* Open non template.yaml file, right click on editor tab, expect to not see menu item
* Open non template.yaml file, right click within file, expect to not see menu item

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] Local run of `gradlew check` succeeds
- [x] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/actions/DeployServerlessApplicationAction.kt [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import com.intellij.openapi.actionSystem.PlatformDataKeys
+import com.intellij.openapi.ui.Messages
+import icons.AwsIcons
+import software.aws.toolkits.resources.message
+
+class DeployServerlessApplicationAction : AnAction(
+        message("serverless.application.deploy"),
+        null,
+        AwsIcons.Resources.LAMBDA_FUNCTION) {
+    private var templateYamlRegex = Regex("template.y[a]?ml", RegexOption.IGNORE_CASE)
```
Comment (abrooksv) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/actions/DeployServerlessApplicationAction.kt]: Technically this matches `template$yaml`, regex also seems kinda wrong tool for the job

File: jetbrains-core/resources/META-INF/plugin.xml [hunk 2]
```diff
@@ -118,6 +118,18 @@
         </group>
         <action class="software.aws.toolkits.jetbrains.core.SettingsSelectorAction" id="aws.settings.selector" icon="AwsIcons.Logos.AWS"/>
         <action class="software.aws.toolkits.jetbrains.core.credentials.CreateOrUpdateCredentialProfilesAction" id="aws.settings.upsertCredentials" icon="AwsIcons.Logos.AWS"/>
+
+        <group id="aws.toolkit.serverless.template">
+            <add-to-group group-id="ProjectViewPopupMenu" anchor="before"
```
Comment (abrooksv) [inline jetbrains-core/resources/META-INF/plugin.xml]: I'm not sure what happens when GitHub plugin isnt installed since this action id wont exist

Comment (awschristou) [reply jetbrains-core/resources/META-INF/plugin.xml]: Me either - I'll change this to use the bottom of the menu

Comment (awschristou) [reply jetbrains-core/resources/META-INF/plugin.xml]: image

**Pull request merged**